src/app/components/Tag/Tag.tsx

```diff
@@ -3,25 +3,28 @@ import type { ReactNode } from 'react';
 import styled from 'styled-components';
 
 type TagProps = {
+  id: number;
   children: ReactNode;
   active?: boolean;
   count?: string;
+  onClick: () => void;
 };
 
 export default function Tag({
   children,
   active = false,
   count,
+  onClick,
 }: TagProps): JSX.Element {
   return (
-    <TagStyleActive active={active}>
+    <TagStyleActive active={active} onClick={onClick}>
       {children} {count && `(${count})`}
     </TagStyleActive>
   );
 }
 const TagStyleActive = styled.span<Partial<TagProps>>`
```

Comment:
It's not required to have `Partial` here if you remove `id` and make `onClick` optional.

```suggestion
const TagStyleActive = styled.span<TagProps>`
```